src/config.rs

```diff
@@ -0,0 +1,233 @@
+use std::{
+    fs,
+    env,
+    path::Path,
+    sync::{Arc, Mutex},
+};
+use std::fs::File;
+use std::io::Write;
+use std::process::exit;
+use regex::Regex;
+use serde::Deserialize;
+use toml::value;
+
+/// The response that is injected into a panic, if the config file was configured falsely
+const PANIC_RESPONSE: &str = "Please create a config file yourself or try setting the environment CONFIG_FILE to valid file location!";
+
+/// The versions of the config this program is compatible with
+const COMPATIBLE_VERSIONS: [(u32, u32); 1] = [
+    (0, 2),
+];
+
+lazy_static! {
+    ///
+    /// The global, thread safe configuration of all of this bot.
+    ///
+    /// Examples:
+    /// ```
+    ///
+    /// let config_arc = Arc::clone(&CONFIG);
+    /// let mut config_lock = config_arc.lock();
+    /// while config_lock.is_err() {
+    ///     dbg!("Could not acquire config lock. Waiting...");
+    ///     sleep(Duration::from_millis(5));
+    ///     config_lock = config_arc.lock();
+    /// }
+    /// let value = match config_lock {
+    ///     Ok(config) => {
+    ///         // ACCESS CONFIG FIELDS HERE AND COPY THEM INTO VALUE
+    ///         // Example: String::from(&config.autokommentator.token)
+    ///     },
+    ///     Err(_) => {
+    ///         println!("Something went wrong internally...");
+    ///         return
+    ///     }
+    /// };
+    /// ```
+    ///
+    pub static ref CONFIG: Arc<Mutex<Config>> = Arc::new(Mutex::new(Config::new()));
+}
+
+#[derive(Deserialize)]
+/// The default configuration struct that holds the global configuration structure
+pub struct Config {
+    /// The version this config file was created with
+    pub version: String,
+    /// Holds configuration for the Autokommentator bot
+    pub autokommentator: Autokommentator,
+    /// Holds configuration for the Känguru Knecht bot
+    pub kaenguru: Kaenguru,
+}
+
+#[derive(Deserialize)]
+/// Structures the data used by the Känguru Knecht bot
+pub struct Kaenguru {
+    /// The token that is used to log into discord
+    pub token: Option<String>,
+    /// The replies and messages that this bot should react to.
+    pub replies: Vec<Response>
```

Comment:
I think it would be more consistent if this field was named `responses`.
```suggestion
    pub responses: Vec<Response>
```

Reply:
+1

Reply:
9d83b09daca5fd6b7e3892f83d855f06bb94df03: Hope you are happy